static/js/components/ProfileSelect.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { FC, Fragment } from "react";
+import {
+  Button,
+  Col,
+  Icon,
+  Label,
+  List,
+  Row,
+} from "@canonical/react-components";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "../util/queryKeys";
+import { NotificationHelper } from "../types/notification";
+import { fetchProfiles } from "../api/profiles";
+import { LxdProfile } from "../types/profile";
+
+interface Props {
+  notify: NotificationHelper;
+  selected: string[];
+  setSelected: (profiles: string[]) => void;
+}
+
+const ProfileSelect: FC<Props> = ({ notify, selected, setSelected }) => {
+  const toggleProfile = (item: LxdProfile) => {
+    const newSelection = selected.includes(item.name)
+      ? selected.filter((i) => i !== item.name)
+      : [...selected, item.name];
+    setSelected(newSelection);
+  };
+
+  const moveUp = (item: LxdProfile) => {
+    const index = selected.indexOf(item.name);
+    if (index === 0) {
+      return;
+    }
+    const newSelection = [...selected];
+    newSelection.splice(index, 1);
+    newSelection.splice(index - 1, 0, item.name);
+    setSelected(newSelection);
+  };
+
+  const moveDown = (item: LxdProfile) => {
+    const index = selected.indexOf(item.name);
+    if (index === selected.length) {
+      return;
+    }
+    const newSelection = [...selected];
+    newSelection.splice(index, 1);
+    newSelection.splice(index + 1, 0, item.name);
+    setSelected(newSelection);
+  };
+
+  const { data: projectProfiles = [], error } = useQuery({
```

Comment:
Why `projectProfiles` instead of `profiles`?

Reply:
There was some clash during dev with the profiles name, as it was used for other stuff. I'll simplify it now -- thats much better!